openapi.yaml

```
@@ -477,243 +473,146 @@ paths:
-            | LD17 | TIROIR_ZAPM                                                                            |
-            | LD18 | ADDUCTION_OC                                                                           |
-
-        "@type":
-          readOnly: false
-        "@baseType":
-          default: Malfacon
-          const: Malfacon
-        severity:
-          description: |
-            The severity of the malfacon
-
-            Allowed values are:
-
-            * `CRITICAL`
-
-              Si la malfaçon représente un danger grave est imminent pour les personnes,
-              l'OI se réserve le droit d'effectuer la reprise ou à minima sécurise les lieux
-              et demande à l'OC d'intervenir.
-
-              De son côté, l'OC se doit de signaler à l'OI toute malfaçon
-              représentant un danger grave et imminent pour les personnes.
-
-            * `MAJOR`
-
-              La malfaçon signifie une détérioration du site technique ou un risque d'intéruption
-              massive de service (dérangement collectif) ou un empêchement à produire
-              de nouveaux client (depuis le passage de commande jusqu'au raccordement).
-
-              Exemples:
-
-              * défaut d'étanchéité sur PB
-              * PM ouvert
-              * site insalubre
-              * brassage encombrant
-              * données de localisation erronées
-
-            * `MINOR`
-
-              Les autres non conformités à corriger.
-          enum:
-            - CRITICAL
-            - MAJOR
-            - MINOR
-        status:
-          description: |
-            Status de la malfacon.
+        - discriminator:
+            propertyName: "@type"
+            mapping:
+              "malfacon.pm.oi": "#/components/schemas/MalfaconOiPm"
+              "malfacon.oi.amontpm": "#/components/schemas/MalfaconOiAmontPm"
+              "malfacon.pbo.oi": "#/components/schemas/MalfaconOiPbo"
+              "malfacon.ccf.oi": "#/components/schemas/MalfaconOiCcf"
+              "malfacon.pm.oc": "#/components/schemas/MalfaconOcPm"
+              "malfacon.amontpm.oc": "#/components/schemas/MalfaconOcAmontPm"
+              "malfacon.pmpbo.oc": "#/components/schemas/MalfaconOcPmPbo"
+              "malfacon.pbo.oc": "#/components/schemas/MalfaconOcPbo"
+              "malfacon.ccf.oc": "#/components/schemas/MalfaconOcCcf"
+          type: object
+          required:
+            - code_oi
+            - code_oc
+            - "@type"
+            - severity
+            - quantity
+          properties:
+            code_oi: #Code ARCEP de l'OI.
+              $ref: https://raw.githubusercontent.com/before-interop/common/v1/common/schemas/Interop.openapi.yaml#/components/schemas/CodeOperateurArcep
+            code_oc: #Code ARCEP de l'OC.
+              $ref: https://raw.githubusercontent.com/before-interop/common/v1/common/schemas/Interop.openapi.yaml#/components/schemas/CodeOperateurArcep
```

Je trouve ca dommage d'avoir un object intermediaire pour juste avoir une string 

```suggestion
            code_oi: #Code ARCEP de l'OI.
              allOf: 
              - $ref: https://raw.githubusercontent.com/before-interop/common/v1/common/schemas/Interop.openapi.yaml#/components/schemas/CodeOperateurArcep
            code_oc: #Code ARCEP de l'OC.
              allOf: 
              - $ref: https://raw.githubusercontent.com/before-interop/common/v1/common/schemas/Interop.openapi.yaml#/components/schemas/CodeOperateurArcep
```